### ISSUE TITLE
Equivalent types reported as not equal

### ISSUE THREAD

**Issue by ethframe** [microsoft/pyright#2876]
**Describe the bug**
In some cases pyright reports the same type variables as not equal.

**To Reproduce**
```python
from typing import List, TypeVar, Union

V = TypeVar("V")
U = TypeVar("U")

Alias = Union[V, U]


def fn(x: Alias[V, V]) -> V:
    return x


def fn2(x: List[Alias[V, V]]) -> List[V]:
    return x
```

```
$ pyright bug.py
...
...\bug.py
  ...\bug.py:10:12 - error: Expression of type "V@fn" cannot be assigned to return type "V@fn"
    Type "V@fn" cannot be assigned to type "V@fn" (reportGeneralTypeIssues)
  ...\bug.py:14:12 - error: Expression of type "List[V@fn2]" cannot be assigned to return type "List[V@fn2]"
    TypeVar "_T@list" is invariant
      Type "V@fn2" cannot be assigned to type "V@fn2" (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 infos
```

**Expected behavior**
No errors.

**VS Code extension or command-line**
pyright 1.1.209

**Additional context**
This is possibly a regression introduced in 1.1.160. pyright 1.1.159 reports no errors in the code sample. Mypy also reports no errors in the code sample.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.210, which I just published. It will also be included in the next release of pylance.
